File: java/src/com/recursion/ibh/Sort.java
package com.recursion.ibh;

import java.util.List;

public class Sort {

    public void sort(List<Integer> arr) {
        if (arr.size() == 1) {
            return;
        }

        int temp = arr.get(arr.size() - 1);
        arr.remove(arr.size() - 1);
        sort(arr);
        insert(arr, temp);
    }


    private void insert(List<Integer> list, int temp) {
        if (list.isEmpty() || list.get(list.size() - 1) <= temp) {
            list.add(temp);
            return;
        }
        int temp1 = list.get(list.size() - 1);
        list.remove(list.size() - 1);
        insert(list, temp);
        list.add(temp1);
    }
}

File: js/recursion/ibh/sort.js
const insert = (arr, temp) => {
  if (arr.length == 0 || arr[arr.length - 1] <= temp) {
    arr.push(temp);
    return;
  }
  let temp1 = arr[arr.length - 1];
  arr.pop();
  insert(arr, temp);
  arr.push(temp1);
};

export const sort = (arr) => {
  if (arr.length == 1) {
    return;
  }
  let temp = arr[arr.length - 1];
  arr.pop();
  sort(arr);
  insert(arr, temp);
};
